### ISSUE TITLE
doas become-method hangs forever

### ISSUE THREAD

**Issue by 8191** [ansible/ansible#13449]
When executing commands using the doas become method, the execution seems to hang forever.

``` yaml

---
- hosts: all
  gather_facts: False
  tasks:
    - command: /usr/bin/whoami
      become: True
```

```
% ansible-playbook -i test.inv -vvv --become-method=doas --ask-become-pass test.yml
No config file found; using defaults
DOAS password:
1 plays in test.yml

PLAY ***************************************************************************

TASK [command] *****************************************************************
task path: /home/mf/cm/test/test.yml:5
<hanoi> ESTABLISH SSH CONNECTION FOR USER: None
<hanoi> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/mf/.ansible/cp/ansible-ssh-%h-%p-%r -tt hanoi ( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1449355062.25-183864280109478 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1449355062.25-183864280109478 )" )
<hanoi> PUT /tmp/tmpylGAdK TO /home/mf/.ansible/tmp/ansible-tmp-1449355062.25-183864280109478/command
<hanoi> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/mf/.ansible/cp/ansible-ssh-%h-%p-%r [hanoi]
<hanoi> ESTABLISH SSH CONNECTION FOR USER: None
<hanoi> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/home/mf/.ansible/cp/ansible-ssh-%h-%p-%r -tt hanoi /bin/sh -c 'doas  -u root  echo BECOME-SUCCESS-lelyicqshqztskrukvdaqmskgruxowwj && doas  -u root  env ANSIBLE=true LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/local/bin/python2.7 /home/mf/.ansible/tmp/ansible-tmp-1449355062.25-183864280109478/command; rm -rf "/home/mf/.ansible/tmp/ansible-tmp-1449355062.25-183864280109478/" > /dev/null 2>&1'
```

How to debug this the best way?

My `/etc/doas.conf`:

```
permit keepenv { PKG_PATH ENV PS1 SSH_AUTH_SOCK } :wheel
permit nopass keepenv { PKG_PATH ENV PS1 SSH_AUTH_SOCK } root
```

Using ansible 955710267c1992c5e3b5b9eb77f4c76e289e3313

**Comment by bcoca** [comment]
I reproduced, but it only seems to be an issue when passwords are required

**Comment by bcoca** [comment]
so the issue is that we execute doas twice, which prompts for passwords twice, but we only supply one

**Comment by bcoca** [comment]
can you verify that the PR above fixes your issue?

**Comment by 8191** [comment]
Yes, works for me.

**Comment by bcoca** [comment]
closing via merged #13451
